validator/coordinator/src/main/java/tech/pegasys/teku/validator/coordinator/performance/DefaultPerformanceTracker.java

```diff
@@ -232,12 +232,17 @@ private AttestationPerformance getAttestationPerformanceForEpoch(
 
       // Check if the attestation had correct target
       Bytes32 attestationTargetRoot = sentAttestation.getData().getTarget().getRoot();
-      if (attestationTargetRoot.equals(get_block_root(state, analyzedEpoch))) {
+      if (attestationTargetRoot.equals(
+          specProvider.at(state).getBeaconStateUtil().getBlockRoot(state, analyzedEpoch))) {
         correctTargetCount++;
 
         // Check if the attestation had correct head block root
         Bytes32 attestationHeadBlockRoot = sentAttestation.getData().getBeacon_block_root();
-        if (attestationHeadBlockRoot.equals(get_block_root_at_slot(state, sentAttestationSlot))) {
+        if (attestationHeadBlockRoot.equals(
+            specProvider
+                .at(state)
+                .getBeaconStateUtil()
```

Comment:
nit: might be worth extracting `BeaconStateUtil` as a variable.

Reply:
i've migrated this to the single level calls, see what you think on the latest version...